js/hotgraphicView.js

```diff
@@ -0,0 +1,223 @@
+                this.remove();
+            }, this));
+        },
+
+        prepareNarrativeModel: function() {
+            var model = this.model;
+            model.set({
+                '_component': 'narrative',
+                '_wasHotgraphic': true,
+                'originalBody': model.get('body'),
+                'originalInstruction': model.get('instruction')
+            });
+
+            // check if active item exists, default to 0
+            var active = model.getActiveItem();
+            if (!active) {
+                model.getItem(0).toggleActive(true);
+            }
+            
+            if (model.get('mobileBody')) {
+                model.set('body', model.get('mobileBody'));
+            }
+            if (model.get('mobileInstruction')) {
+                model.set('instruction', model.get('mobileInstruction'));
+            }
+            return model;
+        },
+
+        onItemsActiveChange: function(model, _isActive) {
+            var selector = 'item-'+model.get('_index');
+            this.$('.hotgraphic-graphic-pin.'+selector).toggleClass('active', _isActive);
+        },
+        
+        onItemsVisitedChange: function(model, _isVisited) {
+            if (_isVisited) {
+                var selector = 'item-'+model.get('_index');
+                this.$('.hotgraphic-graphic-pin.'+selector).addClass('visited');
+            }
+        },
+
+        onPinClicked: function (event) {
+            if(event) {
+                event.preventDefault();
+            }
+            this.selectedPin = event.currentTarget;
+            var $currentHotSpot = $(this.selectedPin);
+            $currentHotSpot.show().addClass('active');
+            this.setVisited($currentHotSpot.data('index'));
+            this.model.set('_isPopupOpen', true);
+        },
+        
+        openPopup: function(model, _isPopupOpen) {
+            if (!_isPopupOpen) {
+                Adapt.trigger('notify:close');
+                return;
+            }
+
+            this.popupView = new HotgraphicPopupView({
+                model: this.model
+            });
+
+            Adapt.trigger("notify:popup", {
+                _view: this.popupView,
+                _isCancellable: true,
+                _showCloseButton: false,
+                _closeOnBackdrop: true,
+                _classes: ' hotgraphic'
+            });
+
+            Adapt.trigger('popup:opened',  $(this.selectedPin));
+
+            this.listenTo(this.popupView, {
+                'popup:closed': this.onPopupClosed,
+                'popup:previous': this.onPopupPrevious,
+                'popup:next': this.onPopupNext
+            });
+
+            this.$('.hotgraphic-popup-inner .active').a11y_focus();
+        },
+
+        onPopupClosed: function() {
+            this.model.getActiveItem().set('_isActive', false);
+            this.removePopupEvents();
+            this.model.set('_isPopupOpen', false);
+            Adapt.trigger('popup:closed');
```

Comment:
`Adapt.trigger('popup:closed');` is unneeded as notify does this itself [here](https://github.com/adaptlearning/adapt_framework/blob/master/src/core/js/views/notifyView.js#L226)